cider/tests/CiderNewHashTableTest.cpp

```diff
@@ -0,0 +1,265 @@
+        smallint_ptr[i] = i;
+        int_ptr[i] = i;
+        bigint_ptr[i] = i;
+        float_ptr[i] = i;
+        double_ptr[i] = i;
+        bool_true_ptr[i] = true;
+        bool_false_ptr[i] = false;
+      }
+
+      std::vector<int8_t*> col_datas = {reinterpret_cast<int8_t*>(tinyint_ptr),
+                                        reinterpret_cast<int8_t*>(tinyint_ptr),
+                                        reinterpret_cast<int8_t*>(smallint_ptr),
+                                        reinterpret_cast<int8_t*>(smallint_ptr),
+                                        reinterpret_cast<int8_t*>(int_ptr),
+                                        reinterpret_cast<int8_t*>(int_ptr),
+                                        reinterpret_cast<int8_t*>(bigint_ptr),
+                                        reinterpret_cast<int8_t*>(bigint_ptr),
+                                        reinterpret_cast<int8_t*>(float_ptr),
+                                        reinterpret_cast<int8_t*>(float_ptr),
+                                        reinterpret_cast<int8_t*>(double_ptr),
+                                        reinterpret_cast<int8_t*>(double_ptr),
+                                        reinterpret_cast<int8_t*>(bool_true_ptr),
+                                        reinterpret_cast<int8_t*>(bool_true_ptr),
+                                        reinterpret_cast<int8_t*>(bool_false_ptr),
+                                        reinterpret_cast<int8_t*>(bool_false_ptr)};
+
+      MockTableForTest = new MockTable(col_names, col_types, col_datas, element_num);
+    }
+  }
+
+  static void TearDownTestSuite() {
+    delete MockTableForTest;
+    MockTableForTest = nullptr;
+  }
+};
+
+MockTable* CiderNewHashTableTest::MockTableForTest = nullptr;
+
+struct Block {
+  int64_t sum = 0;
+  size_t count = 0;
+
+  void add(int8_t value) {
+    sum += value;
+    ++count;
+  }
+
+  template <size_t unroll_count = 128 / sizeof(int8_t)>
+  void addBatch(const int8_t* ptr, size_t size) {
+    /// Compiler cannot unroll this loop, do it manually.
+    /// (at least for floats, most likely due to the lack of -fassociative-math)
+
+    int8_t partial_sums[unroll_count]{};
+
+    const auto* end = ptr + size;
+    const auto* unrolled_end = ptr + (size / unroll_count * unroll_count);
+
+    while (ptr < unrolled_end) {
+      for (size_t i = 0; i < unroll_count; ++i)
+        partial_sums[i] += ptr[i];
+      ptr += unroll_count;
+    }
+
+    for (size_t i = 0; i < unroll_count; ++i)
+      sum += partial_sums[i];
+
+    while (ptr < end) {
+      sum += *ptr;
+      ++ptr;
+    }
+
+    count += size;
+  }
+
+  void merge(const Block& other) {
+    sum += other.sum;
+    count += other.count;
+  }
+
+  int64_t getSum() const { return sum; }
+
+  size_t getCount() const { return count; }
+
+  double getAvg() const { return (double)sum / count; }
+
+  bool operator!() const { return !count; }
+};
+
+TEST_F(CiderNewHashTableTest, aggInt8Test) {
```

Comment:
I think this test is not verifying hashmap functionality, it's more like an example about how to do aggregation based on this hashmap.

Reply:
Yes, these are examples of using `FixedImplicitZeroHashMap` and `FixedImplicitZeroHashMapWithCalculatedSize`.